portal/src/main/webapp/js/src/oncoprint/webgl/setup.js

```diff
@@ -1734,6 +1734,46 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		    }
 		    oncoprint.removeTracks(tracks_to_remove);
 		});
+		
+		$(toolbar_selector + ' #oncoprint_diagram_heatmap_menu #cluster_heatmaps_btn').click(function (evt) {
+		    // suppress dropdown hiding and form submit
+		    evt.preventDefault();
+		    evt.stopPropagation();
+
+		    // cluster heatmap for selected profile:
+		    var genetic_profile_id = $(toolbar_selector + ' #oncoprint_diagram_heatmap_menu #oncoprint_diagram_heatmap_profiles').val();
+		    if (genetic_profile_id) {
+				//sort according to order found in the clustering results: 
+				State.sorting_by_given_order = true;
+		    	var grp = State.heatmap_track_groups[genetic_profile_id];
+				var heatmap_track_group_id = grp.track_group_id; 
```

Comment:
what you're calling `track_uid_map` and computing in `getTrackUIDMap` is actually available identically as `grp.gene_to_track_id`, as far as I can tell.